share/eds/store.go

```diff
@@ -197,6 +201,12 @@ func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.Ext
 }
 
 func (s *Store) put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) (err error) {
+	if _, exists := s.skip.LoadOrStore(root.String(), 1); exists {
+		// no need to do any work, another routine is putting eds for the same root
+		return nil
```

Comment:
Another potential issue is that Put will return immediately for other routines without making them wait until the operation is actually finished. This is problematic if Put fails for any reason keeping other routines in false state.

Reply:
Thats a point! Changed skip to lock. Also some routines, like rpc get call, don't really need to wait for Put to succeed and should resolve regardless.